### PR TITLE
New API request search for query 

### PR DESCRIPTION
**Pull request by MirekSobota**
I have added a new API request that will search for movies in the database if the user enters a title in the search input.

Please check it and let me know what you think and what needs to be improved.

Edit:
We have added some changs in to this PR please take a look and approve if it ok!

### CODE REVIEW

**Review by PawelOwiesek — COMMENTED** [review]
The code seems to be fine, I'm wondering if I should add a validation check of the downloaded data, e.g. in page:
 if (!page || typeof page!== 'number' || page <= 0) {
    throw new Error('Invalid page.');
  }

File: src/features/movieListSlice.js
Comment (PawelOwiesek) [inline src/features/movieListSlice.js:1]: I think it's ok, but there seems to be formatting differences between the old files and the new ones.

File: src/features/movieListSaga.js
Comment (PawelOwiesek) [inline src/features/movieListSaga.js:1]: Logic is fine

File: src/features/MovieList/index.js
Comment (PawelOwiesek) [inline src/features/MovieList/index.js:1]: I think the code is logical and simple

**Review by PawelOwiesek — APPROVED** [review]
The logic is correct and simple, everything works as it should. Great job Mirek

File: src/features/MovieList/index.js [hunk 1]
```diff
@@ -7,20 +7,36 @@ import {
...
 
 const MovieList = () => {
   const popularMovies = useSelector(selectMoviesList);
   const status = useSelector(selectStatus);
   const dispatch = useDispatch();
   const page = useSelector(selectPage);
+  const location = useLocation();
+  const query = new URLSearchParams(location.search).get(searchQueryParamName);
+ 
+  const debouncedLoad = useCallback(
```
Comment (PawelOwiesek) [inline src/features/MovieList/index.js:27]: try hier this:const debouncedLoad = useCallback(
  debounce((query) => {
    dispatch(fetchSearchMoviesLoad(query));
  }, 1000),
  [dispatch, fetchSearchMoviesLoad]
);

**Review by DorotaZielinska — APPROVED** [review]

File: src/features/AsideActions/Loading/loading.js
Comment (DorotaZielinska) [inline src/features/AsideActions/Loading/loading.js:1]: looks ok

File: src/features/MovieList/index.js
Comment (DorotaZielinska) [inline src/features/MovieList/index.js:1]: Great work, you have added NotFound&Loading is clear to read.

File: src/features/AsideActions/Loading/loading.js
Comment (MirekSobota) [reply src/features/AsideActions/Loading/loading.js:1]: Thanks!

**Review by DorotaZielinska — APPROVED** [review]

**Review by PawelOwiesek — APPROVED** [review]
All changes are approved.

File: src/features/MovieList/index.js
Comment (MirekSobota) [reply src/features/MovieList/index.js:1]: Thanks, we did good job!

**Comment by MirekSobota** [comment]
Good job team! 
Paweł i think that you have good idea with validation, we should add it with next PR.

**Pull request merged**